### ISSUE TITLE
Development Pipeline Slowing: Image Pipeline

### ISSUE THREAD

**Issue by joepavitt** [flowforge/website#632]
### Current Behavior

The local development pipeline (and production build time) are increasingly slowing, this hangs for sometime on the introduced Image pipeline work which massively improves performance when the site is running, but the development experience suffers.

Want to understand, is this slow because:
1. We have _too many_ images?
2. We have too many _big_ images?

Is there anything short term we can look at to improve this?

Raised by https://github.com/flowforge/website/issues/609 and https://github.com/flowforge/website/issues/621 referencing speed of recovery for the server.

### Expected Behavior

Ideally, smooth happy fast times when it comes to developing locally and iterating.

**Comment by joepavitt** [comment]
@Pezmc you'll likely have the best insight here - any quick ideas?

**Comment by ZJvandeWeg** [comment]
On the development side I never understand _why_ we care about large images. When we push the bot will fix it for us. So When I'm developing, leave me alone?

**Comment by joepavitt** [comment]
> However, it should be caching the images, so after first build, each subsequent build should be significantly faster. It sounds like this isn't working? Or you're running the build from a clean directory with no caching each time.

I'm running same directory every time, and I know others in the team (cc @iskerrett and @Yndira-FlowForge ) are also hitting this issue a lot too.

**Comment by Pezmc** [comment]
I've just tested this locally and can confirm the caching is behaving as expected, even with all the Netlify changes.

First build (via `npm run start`) after an `npm run clean` took 57 seconds.
Killing the server, then re-running `npm run start` took 13 seconds (12 on a re-run).

However, Netlify appears to automatically open a browser to `http://localhost:8080/` before 11ty has finished building, which _is_ confusing, since the user sees a 404. 11ty hasn't finished the build yet, so I imagine that's most likely causing the confusion here.

I've been able to partially improve this by updating `npm run start` not to do a full clean:

First build (via `npm run start`) after an `npm run clean` took 54 seconds.
Killing the server, then re-running `npm run start` took 12 seconds (10 on a re-run).

However, of note, the page no longer 404's when the server builds, as the **previous build** of 11ty is still available. 
The caveat, is that any page changes won't be reflected until the build completes, which could be _more confusing_

Alternatively, I can tweak the image pipeline not to run at all in dev, however, that means we're seeing different behaviour on prod/dev and I know there are quirks with the `<picture>` and `<srcset>` tags.

**Comment by Pezmc** [comment]
This is partially addressed in #683, let's evaluate in a couple of days how we're feeling.

Possible follow ups, as per that PR:

 - Disable image pipeline entirely locally (speeds up first cold build) - downside possible differences between dev and prod

Would need to ensure dev mode behaves as similarly to prod as possible.

**Comment by Pezmc** [comment]
@joepavitt @iskerrett @robmarcer How have you found working on the website for the last week. Has it been sped up to a more usable speed?

**Comment by robmarcer** [comment]
Thanks @Pezmc, I tend to do something else while I wait so I don't have any useful anecdotes. 

As you said above, the issue I often see now is the dev site isn't actually available even when the terminal window suggests it should be. Now I know it's going to happen it's not that much of a problem, I just don't try for 30 seconds.

**Comment by ZJvandeWeg** [comment]
@robmarcer @Pezmc That indicates that flow state (https://www.medicalnewstoday.com/articles/flow-state#what-it-is) is interrupted, and it's likely still too slow?

**Comment by Pezmc** [comment]
> As you said above, the issue I often see now is the dev site isn't actually available even when the terminal window suggests it should be

 Yes, unfortunately I can't change the wording that Netlify gives us. Though since the changes above, the cache is never cleared, so the server is running and serving as soon as the prompt comes up just serving the old files:
 
 <img width="690" alt="Screenshot 2023-05-03 at 21 25 33" src="https://user-images.githubusercontent.com/507155/235914953-a0c8ee30-6c71-4448-9987-661c8ba0d9b5.png">
 
 The important line is `[11ty] Watching…`, that's when 11ty, which builds the code, is _actually_ ready.

**Comment by Pezmc** [comment]
@ZJvandeWeg Let's leave this issue open then for the follow ups in https://github.com/flowforge/website/issues/632#issuecomment-1524965843

**Comment by Pezmc** [comment]
I think I've found a trick to not build images locally, but still have the output appear the same, am experimenting in down time.
